File: ios/Sources/DeeplinkSDK/DeeplinkLogger.swift
import Foundation

/// Internal debug logger. Enable via `Deeplink.setDebug(true)`.
internal enum DeeplinkLogger {
    static var isEnabled = false

    static func log(_ message: String) {
        guard isEnabled else { return }
        print("[Deeplink] \(message)")
    }

    static func error(_ message: String, _ err: Error? = nil) {
        guard isEnabled else { return }
        let suffix = err.map { " — \($0)" } ?? ""
        print("[Deeplink] ❌ \(message)\(suffix)")
    }
}

File: android/deeplinkSDK/src/main/kotlin/com/deeplink/sdk/DeeplinkLogger.kt
package com.deeplink.sdk

import android.util.Log

/**
 * Internal debug logger. Enable via `DeeplinkSDK.setDebug(true)`.
 * All logs are tagged "[Deeplink]" and use Android's `Log` API.
 */
internal object DeeplinkLogger {
    @Volatile var isEnabled = false
    private const val TAG = "DeeplinkSDK"

    fun log(message: String) {
        if (isEnabled) Log.d(TAG, "[Deeplink] $message")
    }

    fun error(message: String, throwable: Throwable? = null) {
        if (isEnabled) Log.e(TAG, "[Deeplink] ❌ $message", throwable)
    }
}
